examples/js/lines/LineSegments2.js

```diff
@@ -60,6 +60,76 @@ THREE.LineSegments2.prototype = Object.assign( Object.create( THREE.Mesh.prototy
 
 		return this;
 
-	}
+	}, 
+  
+  // NJT added:
+	raycast: ( function () {
+
+		var inverseMatrix = new THREE.Matrix4();
+		var ray = new THREE.Ray();
+		var sphere = new THREE.Sphere();
+
+		return function raycast( raycaster, intersects ) {
 
+			var precision = raycaster.linePrecision;
+
+			var geometry = this.geometry;
+			var matrixWorld = this.matrixWorld;
+
+			// Checking boundingSphere distance to ray
+
+			if ( geometry.boundingSphere === null ) geometry.computeBoundingSphere();
+			sphere.copy( geometry.boundingSphere );
+			sphere.applyMatrix4( matrixWorld );
+			sphere.radius += precision;
+
+			if ( raycaster.ray.intersectsSphere( sphere ) === false ) return;
+			//
+
+			inverseMatrix.getInverse( matrixWorld );
+			ray.copy( raycaster.ray ).applyMatrix4( inverseMatrix );
+
+			var localPrecision = precision / ( ( this.scale.x + this.scale.y + this.scale.z ) / 3 );
+			var localPrecisionSq = localPrecision * localPrecision;
+
+			var vStart = new THREE.Vector3();
+			var vEnd = new THREE.Vector3();
+			var interSegment = new THREE.Vector3();
+			var interRay = new THREE.Vector3();
+			var step = ( this && this.isLineSegments ) ? 2 : 1;
+
+      // Always a LineSegments2 geometry, which stores positions NOT in the positions attribute NJT
+      var starts = geometry.attributes.instanceStart;
+      var ends   = geometry.attributes.instanceEnd;
+      
+      for(var i=0;i<starts.count;i++) {
```

Comment:
Please see https://zz85.github.io/mrdoobapproves/ for help correcting your code style.